cdap-security/src/main/java/co/cask/cdap/security/authorization/Authorizer.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.security.authorization;
+
+import co.cask.cdap.proto.id.EntityId;
+import co.cask.cdap.proto.security.Action;
+import co.cask.cdap.proto.security.Principal;
+
+import java.util.Set;
+
+/**
+ * Interface to grant/revoke {@link Principal principals} authorization for {@link Action actions} on
+ * {@link EntityId CDAP entities}.
+ */
+public interface Authorizer extends AuthorizationEnforcer {
+  /**
+   * Grants a principal authorization to perform a set of actions on an entity.
+   *
+   * @param entity the entity on which an action is being performed
+   * @param principal the Principal that performs the actions. This could be a user, group or a role
+   * @param actions the set of actions to grant
+   */
+  void grant(EntityId entity, Principal principal, Set<Action> actions);
+
+  /**
+   * Grants a Principal authorization to perform all actions on an entity.
+   *
+   * @param entity the entity on which an action is being performed
+   * @param principal the Principal that performs the actions. This could be a user, group or a role
+   */
+  void grant(EntityId entity, Principal principal);
+  /**
+   * Revokes a principal's authorization to perform a set of actions on an entity.
+   *
+   * @param entity the entity on which an action is being performed
+   * @param principal the principal that performs the actions. This could be a user, group or a role
+   * @param actions the set of actions to revoke permissions on
+   */
+  void revoke(EntityId entity, Principal principal, Set<Action> actions);
+
+  /**
+   * Revokes a principal's authorization to perform any action on an entity.
+   *
+   * @param entity the entity on which an action is being performed
+   * @param principal the principal that performs the actions. This could be a user, group or a role
+   */
+  void revoke(EntityId entity, Principal principal);
+
+  /**
+   * Revokes all principals' authorization to perform any action on an entity.
+   *
+   * @param entity the entity on which an action is being performed
+   */
+  void revoke(EntityId entity);
```

Comment:
I had the same question. So this will be needed when an user deletes an entities. So we can call this and remove all permissions for all users. We don't need an equivalent grant because we don't think we will have a use case to grant all privilege to all user on an entity.